src/shell/shell_misc.cpp

```diff
@@ -441,17 +441,19 @@ bool DOS_Shell::Execute(char * name,char * args) {
  * return false =>       check for hardware changes in do_command */
 	char fullname[DOS_PATHLENGTH+4]; //stores results from Which
 	char line[CMD_MAXLINE];
-	if(strlen(args)!= 0){
-		if(*args != ' '){ //put a space in front
-			line[0]=' ';line[1]=0;
-			strncat(line,args,CMD_MAXLINE-2);
+	const bool have_args = args && args[0] != '\0';
+	if (have_args) {
+		if (*args != ' ') { // put a space in front
+			line[0] = ' ';
+			line[1] = 0;
```

Comment:
Since you're making changes here anyway, why not use the more correct `'\0'` instead of `0`?

Reply:
Thanks @shermp.

In some cases, where `array[n] = 0;` was sitting on its own, I just thought it was a numerical quantity assignment into a scalar array!   Had no idea it was a string being reset (or terminated mid-point).

So rework these with a could self-documenting string-utility functions.
The changes start at:  **`Add a safe C-string reset utility function`** (and go down from there).
Suggest reviewing the remaider commit-by-commit.